src/plugins/section/schema.js

```diff
@@ -59,6 +59,25 @@ function normalizeSection(change, error) {
         console.warn('Unhandled violation in section:', code)
         break
 
+    case 'child_type_invalid':
+        if (child.object === 'text') {
+            change.wrapBlock('paragraph')
+            return
+        } else if (child.type === 'title') {
+            // This usually happens when title occure in middle of section after changing
+            // block type from paragraph to title.
+
+            // Create new section with given title and remove unused child.
+            const text = child.text
+            change.withoutNormalizing(() => {
+                change.insertSection(text)
+                change.removeNodeByKey(child.key)
+            })
```

Comment:
This will discard any marks from the title.

I think it would be better to implement it as
```js
change.wrapBlockByKey(child.key, 'section')
const section = change.value.document.getParent(child.key)
change.unwrapNodeByKey(section.key)
```
(note that there is purposefully no `withoutNormalizing`; we want normalizations to run after `wrapBlockByKey` so that they move all nodes after title into the new section).